### PR TITLE
initialContent now shows up for Omnibars; arrow keys work on initial items

### PR DESCRIPTION
**Pull request by JKillian**
#### Changes proposed in this pull request:

When `initialContent` was `null`, it was still getting improperly returned by `renderFilteredItems`. This led to bugs in some of the list components and omnibar where the regular list of items wouldn't be displayed when first opened.

Also fixed a related bug where the arrow keys wouldn't work in an omnibar for the initial content.

### CODE REVIEW

**Comment by JKillian** [comment]
I did notice that even with this fix, `initialContent` doesn't seem to be navigable by arrow keys. Didn't get to look into why though

**Comment by JKillian** [comment]
Pushed an update with an additional fix; is there anything else this needs @giladgray?

**Comment by giladgray** [comment]
why no arrow keys? `initialContent` is a renderer escape hatch: it has no relation to `items` or `itemRenderer` so it is completely separate from the `QueryList` arrow key logic.

**Comment by JKillian** [comment]
When the omnibar shows its list of items when first opened (which it failed to do until the `initialContent` bug was fixed in this PR), it should be navigable by the arrow keys. Prior to this PR, the arrow keys wouldn't work on that list of items because there was logic to disable them w/o a query.

However, if you provided `initialContent` to the omnibar, your initial content, like you said, is unrelated to `items`. Thus we shouldn't handle arrow key presses if you have custom `initialContent`

**Comment by giladgray** [comment]
@JKillian i don't think this is a bug. omnibar is supposed to be empty on first open. see Alfred or Mac OS Spotlight. 

see this line in the docs:

```ts
    /**
     * React content to render when query is empty.
     * If omitted, all items will be rendered (or result of `itemListPredicate` with empty query).
==>  * If explicit `null`, nothing will be rendered when query is empty.   // <== expected behavior
     *
     * This prop is ignored if a custom `itemListRenderer` is supplied.
     */
    initialContent?: React.ReactNode | null;
```

**Comment by JKillian** [comment]
@giladgray, the line above the one that you mentioned is the behavior that should happen:

> If omitted, all items will be rendered (or result of `itemListPredicate` with empty query).

Only if a user specifically says `initialContent={null}` should nothing pop up:

> If explicit `null`, nothing will be rendered when query is empty.

**Comment by JKillian** [comment]
In a lot of use cases, I believe users will want to customize `itemListPredicate` so that recently selected omnibar items show up before any query is entered. This is the typical behavior in IDEs and other places that use omnibar-like functionality I believe

File: packages/select/src/components/omnibar/omnibar.tsx [hunk 1]
```diff
@@ -101,9 +101,11 @@ export class Omnibar<T> extends React.PureComponent<IOmnibarProps<T>, IOmnibarSt
     }
 
     private renderQueryList = (listProps: IQueryListRendererProps<T>) => {
-        const { inputProps = {}, isOpen, overlayProps = {} } = this.props;
+        const { initialContent, inputProps = {}, isOpen, overlayProps = {} } = this.props;
         const { handleKeyDown, handleKeyUp } = listProps;
-        const handlers = isOpen && !this.isQueryEmpty() ? { onKeyDown: handleKeyDown, onKeyUp: handleKeyUp } : {};
+        // handle arrow key presses once we have a query or w/ no query if we're not displaying custom initial content
+        const shouldHandleKeys = isOpen && (initialContent == null || !this.isQueryEmpty());
+        const handlers = shouldHandleKeys ? { onKeyDown: handleKeyDown, onKeyUp: handleKeyUp } : {};
```
Comment (giladgray) [inline packages/select/src/components/omnibar/omnibar.tsx]: i think this was fixed in #3130

File: packages/select/src/common/itemListRenderer.ts [hunk 2]
```diff
@@ -57,7 +57,7 @@ export function renderFilteredItems(
     noResults?: React.ReactNode,
     initialContent?: React.ReactNode | null,
 ): React.ReactNode {
-    if (props.query.length === 0 && initialContent !== undefined) {
+    if (props.query.length === 0 && initialContent != null) {
```
Comment (giladgray) [inline packages/select/src/common/itemListRenderer.ts]: this change feels wrong -- the explicit undefined check is supported by the docs.

**Pull request closed without merging**